src/components/forms/PhoneInput.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from "react"
+
+import {
+  Input,
+  InputGroup,
+  InputLeftAddon,
+  InputRightAddon,
+  InputProps as ChakraInputProps,
+  Select,
+} from "@chakra-ui/react"
+
+export type E164Number = string
+
+export enum CountryCode {
+  AU = "AU",
+  US = "US",
+}
+
+export interface CountryData {
+  name: string
+  emoji: string
+  countryCode: string
+  // TODO use template literal type once we've migrated away from tsdx
+  // and outdated prettier
+  callingCode: string
+  allowedNumberLength: number
+  exampleNumber: string
+}
+
+export type Countries = {
+  [key in CountryCode]: CountryData
+}
+
+const countries: Countries = {
+  [CountryCode.AU]: {
+    name: "Australia",
+    emoji: "🇦🇺",
+    countryCode: "AU",
+    callingCode: "+61",
+    allowedNumberLength: 9,
```

Comment:
can we please rename `allowedNumberLength` to `maxNumberLength`? sorry, this is my fault. I think using `max` makes its purpose much clearer